bpfd/src/bpf.rs

```diff
@@ -311,6 +317,18 @@ impl BpfManager {
                     map_pin_path.expect("map_pin_path must be set after successfult load"),
                 )
                 .await?;
+
+                // Update the local copy with the modified map_used_by
+                if let Some(prog) = self.programs.get_mut(&id) {
+                    let map_used_by = match prog.data() {
```

Comment:
If we actually want to ignore the error here (and above) just do 

` let map_used_by = prog.data().ok().map(|data| data.maps_used_by());
`